crates/accelerate/src/sparse_observable.rs

```diff
@@ -998,6 +1206,133 @@ impl SparseObservable {
+                // without an exception, but limitations in PyO3 prevent this at the moment.  See
+                // https://github.com/PyO3/pyo3/issues/4605.
+                return Err(PyTypeError::new_err(format!(
+                    "invalid object for in-place addition of 'SparseObservable': {}",
+                    other.repr()?
+                )));
+            };
+            let other = other.borrow();
+            slf_.check_equal_qubits(&other)?;
+            *slf_ += &other;
+        }
+        Ok(())
+    }
+
+    fn __sub__(&self, other: &Bound<PyAny>) -> PyResult<Py<PyAny>> {
+        let py = other.py();
+        let Some(other) = coerce_to_observable(other)? else {
+            return Ok(py.NotImplemented());
+        };
+        let other = other.borrow();
+        self.check_equal_qubits(&other)?;
+        Ok((self - &other).into_py(py))
+    }
+    fn __rsub__(&self, other: &Bound<PyAny>) -> PyResult<Py<PyAny>> {
+        let py = other.py();
+        let Some(other) = coerce_to_observable(other)? else {
+            return Ok(py.NotImplemented());
+        };
+        let other = other.borrow();
+        self.check_equal_qubits(&other)?;
+        Ok((<&SparseObservable as ::std::ops::Sub>::sub(&other, self)).into_py(py))
+    }
+    fn __isub__(slf_: Bound<SparseObservable>, other: &Bound<PyAny>) -> PyResult<()> {
+        if slf_.is(other) {
+            // This is not strictly the same thing as `a - a` if `a` contains non-finite
+            // floating-point values (`inf - inf` is `NaN`, for example); we don't really have a
+            // clear view on what floating-point guarantees we're going to make right now.
+            slf_.borrow_mut().clear()
+        } else {
+            let mut slf_ = slf_.borrow_mut();
+            let Some(other) = coerce_to_observable(other)? else {
+                // This is not well behaved - we _should_ return `NotImplemented` to Python space
+                // without an exception, but limitations in PyO3 prevent this at the moment.  See
+                // https://github.com/PyO3/pyo3/issues/4605.
+                return Err(PyTypeError::new_err(format!(
+                    "invalid object for in-place subtraction of 'SparseObservable': {}",
+                    other.repr()?
+                )));
+            };
+            let other = other.borrow();
+            slf_.check_equal_qubits(&other)?;
+            *slf_ -= &other;
+        }
+        Ok(())
+    }
+
+    fn __pos__(&self) -> SparseObservable {
+        self.clone()
+    }
+    fn __neg__(&self) -> SparseObservable {
+        -self
+    }
+
+    fn __mul__(&self, other: Complex64) -> SparseObservable {
+        self * other
+    }
+    fn __rmul__(&self, other: Complex64) -> SparseObservable {
+        other * self
+    }
+    fn __imul__(&mut self, other: Complex64) {
+        *self *= other;
+    }
+
+    fn __truediv__(&self, other: Complex64) -> PyResult<SparseObservable> {
```

Comment:
Doing something ridiculous like 
```
SparseObservable.from_label("X") / 1e-200
```
gives coeffs that are `jnf+NaNj` but `(1+0j)/1e-200` gives, as I would expect, just `1e200+0j`. Is this difference in behavior of complex number handling due to how Rust represents numbers?

Reply:
It's because in Rust space, we coerce `1e-200` up to a `Complex64` on entry to the method, whereas Python is doing the same thing as C, and treating `complex / float` by distributing the division over both components.

You can get the same behaviour in Python with `complex(1, 0)/complex(1e-200, 0)`